.github/workflows/buildsCI.yaml

```diff
@@ -400,4 +400,46 @@ jobs:
        cmake --build . --config ${{env.LRS_BUILD_CONFIG}} -- -j4
        ls
   
-  
+  cppcheck:
+    name: cppcheck-test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        
+      - name: Check_API
+        shell: bash
+        run: |
+          cd scripts
+          ./api_check.sh
+          cd ..  
+        
+      - name: Prerun
+        shell: bash
+        run: |
+          sudo add-apt-repository --yes ppa:ubuntu-toolchain-r/test;
+          sudo apt-get update;
+          sudo apt-get install -qq cppcheck;
+      
+      - name: Run
+        shell: bash
+        run: | 
+          cppcheck . --std=c++11 -ithird-party/easyloggingpp -ithird-party/glad -ithird-party/sqlite -ithird-party/imgui -ithird-party/glfw -ithird-party/hidapi -ithird-party/json.hpp --xml &> cppcheck_run.log
```

Comment:
Can we add some `verbose` flag?
To see some info in the run log?

Reply:
--verbose does not add additional error description information, just information on possible run flags that we don't pass here.

Reply:
@Nir-Az  We can add additional checks, but I don't think this is the right time to add them until we clean the major errors.

Default run - error and warnings - log ~180KB
Some checks - performance, portability, missing include - log ~2MB
All checks - style, unused function, information (mostly cppcheck limitations) - log ~6.5MB

Reply:
I think we only want the critical once, null reference, divide by zero.. stuff like that.
The default answer it right?

Reply:
Yes, than I will only run the default for now